ReSwift/CoreTypes/Store.swift

```diff
@@ -85,51 +85,13 @@ open class Store<State: StateType>: StoreType {
         }
     }
 
-    open func subscribe<S: StoreSubscriber>(_ subscriber: S)
-        where S.StoreSubscriberStateType == State {
-            _ = subscribe(subscriber, transform: nil)
-    }
-
-    open func subscribe<SelectedState, S: StoreSubscriber>(
-        _ subscriber: S, transform: ((Subscription<State>) -> Subscription<SelectedState>)?
-    ) where S.StoreSubscriberStateType == SelectedState
-    {
-        // If the same subscriber is already registered with the store, replace the existing
-        // subscription with the new one.
-        if let index = subscriptions.index(where: { $0.subscriber === subscriber }) {
-            subscriptions.remove(at: index)
-        }
-
-        // Create a subscription for the new subscriber.
-        let originalSubscription = Subscription<State>()
-        // Call the optional transformation closure. This allows callers to modify
-        // the subscription, e.g. in order to subselect parts of the store's state.
-        let transformedSubscription = transform?(originalSubscription)
-
-        let subscriptionBox = self.subscriptionBox(
-            originalSubscription: originalSubscription,
-            transformedSubscription: transformedSubscription,
-            subscriber: subscriber
-        )
-
-        subscriptions.append(subscriptionBox)
-
-        if let state = self.state {
-            originalSubscription.newValues(oldState: nil, newState: state)
-        }
-    }
-
-    internal func subscriptionBox<T>(
-        originalSubscription: Subscription<State>,
-        transformedSubscription: Subscription<T>?,
-        subscriber: AnyStoreSubscriber
-        ) -> SubscriptionBox<State> {
-
-        return SubscriptionBox(
-            originalSubscription: originalSubscription,
-            transformedSubscription: transformedSubscription,
-            subscriber: subscriber
+    open func subscription() -> Subscription<State> {
+        let subscription = Subscription<State>(
+            initialState: self.state,
+            automaticallySkipsEquatable: self.subscriptionsAutomaticallySkipRepeats
         )
+        self.subscriptions.append(subscription)
```

Comment:
This means that store.subscription() already inserts a subscription, even though subscribe() is never called, right?

If so, I think we should change that to operate on a pending subscription that is only activated in the store on subscribe(). This is now more like a factory; the side effect can cause troubles. If you do not call append here, you do not run into problems when you grab a handle of the subscription during its transformation and intend to branch off in 2+ directions. That’d be similar to RxSwift.Observalbe stream transformations and its subscribe call, which is a prominent and rather well understood pattern.